src/Assignments/ProblemDetails.tsx

```diff
@@ -109,17 +111,19 @@ export const ProblemDetails: React.FC<ProblemDetailsProps> = ({
                         >
                             {(currentMoment: moment.Moment) => {
                                 let message = '';
-                                if (currentMoment.isBefore(endDate) && (_.isNil(attemptsRemaining) || attemptsRemaining > 0)) {
+                                if (currentMoment.isBefore(endDate) && (_.isNil(attemptsRemaining) || (attemptsRemaining > 0 && !isClosed))) {
                                     message = `Due ${endDate.fromNow()}`;
                                 } else if (currentMoment.isBefore(deadDate)) {
                                     message = `Partial credit expires ${deadDate.fromNow()}`;
                                 } else if (currentMoment.isBefore(solutionsMoment)) {
                                     message = `Solutions available ${solutionsMoment.fromNow()}`;
-                                } else if (isVersionedAssessment && (attemptsRemaining === 0 || isClosed)) {
-                                    message = 'is completed';
-                                } else {
+                                } else if (currentMoment.isAfter(solutionsMoment)) {
                                     message = (isVersionedAssessment) ? 'Time expired for this version' : 'Past due';
+                                    setAttemptsRemaining && setAttemptsRemaining(0);
```

Comment:
This could be more simply written as 
```suggestion
                                    setAttemptsRemaining?.(0);
```